### ISSUE TITLE
libapt versioned dependency

### ISSUE THREAD

**Issue by panlinux** [CanonicalLtd/ubuntu-advantage-client#416]
Turns out there are cases out there where someone might have just trusty-security enabled, not trusty-updates, and thus will not have the apt version we need which, among other things, supports auth.conf.d:
```
apt (1.0.1ubuntu2.22) trusty; urgency=medium

  * apt.dirs: Install auth.conf.d directory (LP: #1818996)
```
Even requiring apt >= 1.0.1ubuntu2.22 isn't enough, though, as that won't upgrade libapt.

Looks like we need a versioned dependency on `libapt-pkg4.12 (>=1.0.1ubuntu2.22)`. That does look a bit odd, though, as shared library dependencies are usually handled automatically. The current ua package does depend in this lib, because of the apt hook, but requires only version 0.8.0 or higher.

**Comment by powersj** [comment]
Per discussions with Steve and others is this actually needed? If not, can we close this?

**Comment by panlinux** [comment]
The discussion with Steve was in the context of the old client package, which just ships a shell script. In that case, it's super odd to add a libapt-pkg4.12 strict dependency to it, even though it's technically correct.

The new client is a more complete solution, and is also more complex, already pulling in many dependencies. I think it will depend on how many installations would be affected by this bug, namely, how many setups have just trusty-security, and have not updated apt to what is in trusty-updates.

The other option, to avoid adding that explicit dependency, would be to have ua itself make sure a recent enough libapt-pkg4.12 package is installed, and consider that a pre-flight check for when enabling repository-based services.

To summarize:
- leave it as is
- add an explicit `Depends: libapt-pkg4.12 (>=1.0.1ubuntu2.22)` to the ua package (that version is for the Trusty case: it's going to be different for other releases)
- have ua itself check if that dep is installed, or available

**Comment by powersj** [comment]
I believe we fixed this with 13a788afdcbe6fede21c47e68f86dd6f650c98d1
